src/renderers/shared/ReactDebugTool.js

```diff
@@ -109,15 +112,18 @@ function beginLifeCycleTimer(debugID, timerType) {
   if (currentFlushNesting === 0) {
     return;
   }
-  warning(
-    !currentTimerType,
-    'There is an internal error in the React performance measurement code. ' +
-    'Did not expect %s timer to start while %s timer is still in ' +
-    'progress for %s instance.',
-    timerType,
-    currentTimerType || 'no',
-    (debugID === currentTimerDebugID) ? 'the same' : 'another'
-  );
+  if (!beginLifeCycleTimerHasWarned) {
```

Comment:
Great, didn't know how `warning`-'s api. Changed it as you suggested in new commit.